### DISCUSSION TITLE
Dev mode much faster now

### DISCUSSION THREAD

**Discussion by R-Bower** [vercel/next.js#38141] [Help]
I just upgraded from v12.1.6 to v12.2.0 and the improvements thus far are fantastic.  

On 12.1.6, refreshing the browser in local dev mode could take up to 10 seconds.  With v12.2.0, it's instant.  Pages seem to load faster as well.  

Am I just seeing things or did speeds actually improve?  I combed through the release notes but couldn't find anything related to this.

**Comment by icyJoseph** [comment]
Perhaps this?

> New @swc/helpers optimizations prevent duplication across bundles, reducing bundle size by around 2KB in a minimal configuration, and more on larger apps

Bundle sizes have impact both in compiling, downloading, parsing, and execution phases, so even a 2KB cut down can be significant in terms of perceived experience. 

The heaviest app I have at the moment, with Next.js 12.1.6, loads in less than 1 second in dev mode, loading slightly over 1000 modules. I reckon for an app that starts at 10 seconds in Next.js 12.1.6, any code optimisation reduces compile, downloading, parsing and execution greatly.
